src/rust/bitbox02-rust-c/src/lib.rs

```diff
@@ -25,16 +25,21 @@ extern crate std;
 #[macro_use]
 mod alloc;
 
+mod util;
+
+#[cfg(feature = "firmware")]
 mod async_usb;
+#[cfg(feature = "firmware")]
 mod noise;
+#[cfg(feature = "firmware")]
 mod sha2;
-mod util;
+#[cfg(feature = "firmware")]
 mod workflow;
 
-#[cfg(feature = "platform-bitboxbase")]
+#[cfg(all(feature = "firmware", feature = "platform-bitboxbase"))]
 pub mod bitboxbase;
 
-#[cfg(feature = "platform-bitbox02")]
+#[cfg(all(feature = "firmware", feature = "platform-bitbox02"))]
 pub mod bitbox02;
```

Comment:
This seems strange. Why would it need both firmware and platform-bitbox02. Wouldn't platform-bitbox02 be enough?

Reply:
there is platform bitbox02 and platform bitboxbase. both have a firmware and a bootloader.

since we don't have any real bootloader rust code, I added the firmware guard in `mod bitboxbase` above, and here too for consistency.

Reply:
oh, I was looking at it in the wrong way. this is deliberate, of course :)